lib/ddtrace/tracer.rb

```diff
@@ -233,6 +233,8 @@ def start_span(name, options = {})
         if ctx && ctx.trace_id && ctx.span_id
           span.trace_id = ctx.trace_id
           span.parent_id = ctx.span_id
+        elsif ctx && ctx.trace_id && ctx.origin == 'synthetics'
```

Comment:
I don't like having a check for synthetics directly here; it conflates the responsibility of the tracer with the synthetics feature, towards which the tracer should be totally apathetic.

Maybe the better way of solving this is at the distributed tracing propagation code? If synthetics is encountered there, then parent ID should be forced to 0? It would then pass the check it was failing on line 233.

Reply:
@delner Thanks for pushing this up as a quick fix. 

One small thing, It appears that the origin can be either `synthetics` or `synthetics-browser` , depending on if the synthetics check is an `api` test or a `browser` test. I would suggest handling this with either an additional case `(ctx.origin == 'synthetics' || ctx.origin == 'synthetics-browser')` or something perhaps a bit more extensible but slightly less performant like `ctx.origin.include?('synthetics')`.

There is probably a larger discussion here of what the actual naming convention is for the origin headers, but this should fit the bill for now 😅 

There is also a check a bit earlier, in the distributed tracing code: https://github.com/DataDog/dd-trace-rb/blob/f3abc8d09ca6ec865ade4bfe920a8a5115307b2e/lib/ddtrace/distributed_tracing/headers/datadog.rb#L31

`return unless (trace_id && parent_id) || (origin == 'synthetics' && trace_id)`

That would need to get adjusted similarly to account for `synthetics-browser` origin

Reply:
Please wait, verifying this change.

Reply:
> then parent ID should be forced to 0?

This seems fine to me, if it makes sense to do. `parent_id = nil` is basically the same as `parent_id = 0`, gets converted to `0` by the agent/backend/whoever anyways

Reply:
@brettlangdon Okay, I will try to change it so that it forces parent ID to 0 here, because `nil` is not the same thing as 0 for Ruby.

@ericmustin This is the sort of issue I was alluding to before; that by synthetics not sending a parent ID header, we can't treat synthetics as normal distributed tracing headers, and the special case considerations will effectively couple this to synthetics implementation in a brittle way (e.g. we now have to update the `origin` check to match its new implementation.)

Ultimately, yeah... `ctx.origin.include?('synthetics')` is probably the simplest solution we could manage right now. Thanks for pointing out this specific issue; didn't know it had changed.

Reply:
Added a test for `synthetics-browser` and changed the check to be more flexible, so I think this should be fixed. (Thanks for the heads up on this @ericmustin!)

I change the nature of how this check works so its more permissive of whatever the context gives it. https://github.com/DataDog/dd-trace-rb/pull/856#discussion_r347620372

We could still change the propagation code to return 0 for the parent ID for synthetics, but it's not strictly necessary now: the propagation code will return `nil`, but the new check will accept  ignore the `nil` parent ID, and retain a 0 value.

Marking this as resolved.